### ISSUE TITLE
💅 `correctness/noUnreachable` incorrect for `try/catch/finally` when there is a jump before the `finally` block

### ISSUE THREAD

**Issue by dylwil3** [biomejs/biome#4946]
### Environment information

```bash
See playground link.
```

### Rule name

noUnreachable

### Playground link

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAZgAoACkAIAB7AAoACQB3AGgAaQBsAGUAIAAoAHQAcgB1AGUAKQAgAHsACgAJAAkAdAByAHkAIAB7AAoACQAJAAkAYgByAGUAYQBrADsACgAJAAkAfQAgAGYAaQBuAGEAbABsAHkAIAB7AAoACQAJAAkAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAIgByAGUAYQBjAGgAYQBiAGwAZQAiACkAOwAKAAkACQB9AAoACQAJAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACIAdQBuAHIAZQBhAGMAaABhAGIAbABlACIAKQA7AAoACQB9AAoAfQA%3D

### Expected result

In the below snippet, both log statements are marked as unreachable, but only the second is. It looks like the control flow analysis is not taking into account that the `finally` block is resolved before any jumps from the `try` block?

```js
function f() {
	while (true) {
		try {
			break;
		} finally {
			console.log("reachable");
		}
		console.log("unreachable");
	}
}
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
